SU2_CFD/src/fluid/CFluidScalar.cpp

```diff
@@ -0,0 +1,277 @@
+        LaminarViscosityPointers[iVar] = unique_ptr<CPolynomialViscosity<N_POLY_COEFFS>>(
+          new CPolynomialViscosity<N_POLY_COEFFS>(config->GetMu_PolyCoeffND()));
+      }
+      break;
+    case VISCOSITYMODEL::FLAMELET:
+      /* do nothing. Viscosity is obtained from the table and set in setTDState_T */
+      break;
+    default:
+      SU2_MPI::Error("Viscosity model not available.", CURRENT_FUNCTION);
+      break;
+  }
+}
+
+void CFluidScalar::SetThermalConductivityModel(const CConfig* config) {
+  switch (config->GetKind_ConductivityModel()) {
+    case CONDUCTIVITYMODEL::CONSTANT:
+      for(int iVar = 0; iVar < n_species_mixture; iVar++){
+        if (config->GetKind_ConductivityModel_Turb() == CONDUCTIVITYMODEL_TURB::CONSTANT_PRANDTL) {
+          ThermalConductivityPointers[iVar] = unique_ptr<CConstantConductivityRANS>(
+            new CConstantConductivityRANS(config->GetKt_Constant(iVar), config->GetPrandtl_Turb(iVar)));
+        } else {
+          ThermalConductivityPointers[iVar] = unique_ptr<CConstantConductivity>(new CConstantConductivity(config->GetKt_Constant(iVar)));
+        }
+      }  
+      break;
+    case CONDUCTIVITYMODEL::CONSTANT_PRANDTL:
+      for(int iVar = 0; iVar < n_species_mixture; iVar++){
+        if (config->GetKind_ConductivityModel_Turb() == CONDUCTIVITYMODEL_TURB::CONSTANT_PRANDTL) {
+          ThermalConductivityPointers[iVar] = unique_ptr<CConstantPrandtlRANS>(
+            new CConstantPrandtlRANS(config->GetPrandtl_Lam(iVar), config->GetPrandtl_Turb(iVar)));
+        } else {
+          ThermalConductivityPointers[iVar] = unique_ptr<CConstantPrandtl>(new CConstantPrandtl(config->GetPrandtl_Lam(iVar)));
+        }
+      }
+      break;
+    case CONDUCTIVITYMODEL::POLYNOMIAL:
+      for(int iVar = 0; iVar < n_species_mixture; iVar++){
+        if (config->GetKind_ConductivityModel_Turb() == CONDUCTIVITYMODEL_TURB::CONSTANT_PRANDTL) {
+          ThermalConductivity = unique_ptr<CPolynomialConductivityRANS<N_POLY_COEFFS>>(
+            new CPolynomialConductivityRANS<N_POLY_COEFFS>(config->GetKt_PolyCoeffND(), config->GetPrandtl_Turb()));
+        } else {
+          ThermalConductivity = unique_ptr<CPolynomialConductivity<N_POLY_COEFFS>>(
+            new CPolynomialConductivity<N_POLY_COEFFS>(config->GetKt_PolyCoeffND()));
+        }
+      }
+      break;
+    case CONDUCTIVITYMODEL::FLAMELET:
+      /* do nothing. Conductivity is obtained from the table and set in setTDState_T */
+      break;
+    default:
+      SU2_MPI::Error("Conductivity model not available.", CURRENT_FUNCTION);
+      break;
+  }
+}
+
+su2double CFluidScalar::wilkeViscosity(su2double* val_scalars){
+  std::vector<su2double> phi; 
+  std::vector<su2double> wilkeNumerator;
+  std::vector<su2double> wilkeDenumeratorSum;
+  su2double wilkeDenumerator = 0.0; 
+  wilkeDenumeratorSum.clear();
+  wilkeNumerator.clear(); 
+  su2double viscosityMixture = 0.0;
+   
+  massToMoleFractions(val_scalars); 
```

Comment:
Right now I only moved `massToMoleFractions` to `SetTDState_T`. I did not make the method `const` because `laminarViscosity` is altered by `wilkeViscosity`. I placed `laminarViscosity` within the function because the `laminarViscosity` value may change at every grid point, e.g. when using Sutherland law in case of non-isothermal flow. Is there a better approach to do this?